lib/ansible/parsing/vault/__init__.py

```diff
@@ -923,6 +925,31 @@ def decrypt_file(self, filename, output_file=None):
             raise AnsibleError("%s for %s" % (to_bytes(e), to_bytes(filename)))
         self.write_data(plaintext, output_file or filename, shred=False)
 
+    def convert_to_inline(self, filename, secret, indent=2, output_file=None):
+        # follow the symlink
+        filename = self._real_path(filename)
+        ciphertext = self.read_data(filename)
+        try:
+            plaintext = self.vault.decrypt(ciphertext, filename=filename)
+        except AnsibleError as e:
+            raise AnsibleError("%s for %s" % (to_bytes(e), to_bytes(filename)))
+        data = yaml.load(plaintext)
```

Comment:
`plaintext` might not actually be YAML. This code was written assuming that it always operates on `vars` files, but we should at least fail in some sensible way when `plaintext` is not YAML

Reply:
vault is also used to encrypt private keys and certs, so you cannot always assume its valid yaml

Reply:
but this is only for the 'inline' function? .. that might be easier to get away with